### Dematel Yöntemi Nedir?
##### - Dematel yöntemi, karmaşık sistemlerdeki etkileşimlerin ve geri bildirimlerin analizinde kullanılan bir yöntemdir. Bu yöntem, bir sistemin unsurları arasındaki nedensellik ilişkilerini ve bunların etkilerini belirlemek için kullanılır.

### Dematel Yönteminin Kullanım Alanları:

##### - Stratejik planlama
##### - Risk analizi
##### - Karar verme
##### - Problem çözme
##### - Sistem tasarımı

#### Dematel Yönteminin Neden Kullanılır ? :

##### - Karmaşık sistemleri analiz etmek için kullanışlı bir araçtır.
##### - Bir sistemin unsurları arasındaki nedensellik ilişkilerini ve bunların etkilerini belirlemeye yardımcı olur.
##### - Görselleştirme yoluyla sistemin daha iyi anlaşılmasını sağlar.

#### Dematel Yönteminin Avantajları:

##### - Kolay uygulanabilir.
##### - Görsel olarak kolay anlaşılır.
##### - Farklı disiplinlerde kullanılabilir.

#### Dematel Yönteminin Dezavantajları:

##### - Karşılaştırma matrisindeki değerlerin öznel olması nedeniyle hatalara açık olabilir.
##### - Büyük sistemler için karmaşık ve zaman alıcı olabilir.

#### Dematel Yönteminin Adımları:

##### - Tanımlama: Dematel yönteminin ilk adımı, analiz edilecek sistemi ve sistemin unsurlarını tanımlamaktır. Bu unsurlar, departmanlar, kişiler, faaliyetler veya sistemin diğer bileşenleri olabilir.

##### - Karşılaştırma Matrisi Oluşturma: Sistemin unsurları arasındaki etkileşimleri ve geri bildirimleri değerlendirmek için bir karşılaştırma matrisi oluşturulur. Bu matriste, her bir unsur diğer her bir unsurla karşılaştırılır ve etkileşiminin derecesi bir sayısal değerle gösterilir.

##### - Doğrudan ve Dolaylı Etki Matrisleri: Karşılaştırma matrisinden doğrudan ve dolaylı etki matrisleri hesaplanır. Doğrudan etki matrisi, bir unsurun diğer unsurlar üzerindeki doğrudan etkisini gösterir. Dolaylı etki matrisi ise bir unsurun diğer unsurlar üzerindeki dolaylı etkisini gösterir.

##### - Toplam Etki Matrisi: Toplam etki matrisi, bir unsurun diğer unsurlar üzerindeki toplam etkisini (doğrudan ve dolaylı etkilerin toplamı) gösterir.

##### - Görselleştirme: Dematel yönteminin sonuçları, bir ağ diyagramı kullanılarak görselleştirilebilir. Bu diyagram, sistemin unsurları arasındaki nedensellik ilişkilerini ve bunların etkilerini gösterir.

#### Dematel Yönteminin Verdiği Sonuçlar:

##### - Doğrudan etki matrisi: Bir unsurun diğer unsurlar üzerindeki doğrudan etkisini gösterir.
##### - Dolaylı etki matrisi: Bir unsurun diğer unsurlar üzerindeki dolaylı etkisini gösterir.
##### - Toplam etki matrisi: Bir unsurun diğer unsurlar üzerindeki toplam etkisini (doğrudan ve dolaylı etkilerin toplamı) gösterir.
##### - Ağ diyagramı: Dematel yönteminin sonuçlarını görselleştiren bir diyagramdır.

#### Dematel Yönteminin Sonuçlarının Yorumlanması:

##### - Toplam etki matrisi: Her bir unsurun diğer unsurlar üzerindeki net etkisini (etki - etkilenen) gösterir.
##### - Ağ diyagramı: Sistemin unsurları arasındaki nedensellik ilişkilerini ve bunların yönlerini gösterir.

#### Dematel Yönteminin Sonuçlarının Kullanımı:

##### - Bir sistemin unsurları arasındaki nedensellik ilişkilerini ve bunların etkilerini daha iyi anlamaya yardımcı olur.
##### - Sistemin hangi unsurlarına odaklanılması gerektiğini belirlemeye yardımcı olur.
##### - Sistemin performansını artırmak için stratejiler geliştirmeye yardımcı olur.

##### ___________________________________________________________________________________________________________
##### ___________________________________________________________________________________________________________
##### ___________________________________________________________________________________________________________
##### ___________________________________________________________________________________________________________

#### Örnek:

##### Bir şirketin 5 departmanı olduğunu varsayalım:

##### Pazarlama: Ürünlerin ve hizmetlerin potansiyel müşterilere tanıtılması ve satılması.
##### Satış: Potansiyel müşterilerle iletişim kurmak ve satışları tamamlamak.
##### Üretim: Ürünlerin tasarlanması, geliştirilmesi ve üretilmesi.
##### Ar-Ge: Yeni ürünler ve hizmetler için araştırma ve geliştirme çalışmaları yapmak.
##### Finans: Şirketin finansal kaynaklarını yönetmek.

In [1]:
import numpy as np
import pandas as pd

In [2]:
data = np.array([
    [0, 3, 2, 4, 1],
    [2, 0, 3, 4, 2],
    [1, 2, 0, 3, 2],
    [4, 3, 2, 0, 1],
    [3, 1, 2, 4, 0]
])
columns = ["Pazarlama", "Satış", "Üretim", "Ar-Ge", "Finans"]
index = ["Pazarlama", "Satış", "Üretim", "Ar-Ge", "Finans"]
df = pd.DataFrame(data, columns=columns, index=index)
df

Pazarlama  Satış  Üretim  Ar-Ge  Finans
Pazarlama          0      3       2      4       1
Satış              2      0       3      4       2
Üretim             1      2       0      3       2
Ar-Ge              4      3       2      0       1
Finans             3      1       2      4       0

In [3]:
df

Pazarlama  Satış  Üretim  Ar-Ge  Finans
Pazarlama          0      3       2      4       1
Satış              2      0       3      4       2
Üretim             1      2       0      3       2
Ar-Ge              4      3       2      0       1
Finans             3      1       2      4       0

In [4]:
D = np.zeros_like(data, dtype=float)
for i in range(data.shape[0]):
    for j in range(data.shape[1]):
        if i != j:
            D[i, j] = data[i, j] - data[j, i]

positives = np.maximum(D, 0)
negatives = np.maximum(-D, 0)
total_positive = np.sum(positives, axis=1)
total_negative = np.sum(negatives, axis=1)
D_positive = total_positive / (total_positive + total_negative)
D_negative = total_negative / (total_positive + total_negative)

In [5]:
print("Departmanlar arasındaki Dematel Pozitif Endeksleri:")
for i, department in enumerate(["Pazarlama", "Satış", "Üretim", "Ar-Ge", "Finans"]):
    print(f"{department}: {D_positive[i]}")

Departmanlar arasındaki Dematel Pozitif Endeksleri:
Pazarlama: 0.5
Satış: 0.75
Üretim: 0.3333333333333333
Ar-Ge: 0.0
Finans: 0.8333333333333334


##### - Pazarlama: Diğer departmanlarla olan olumlu etkileşimi orta düzeydedir (0.5). Bu, pazarlama departmanının diğer departmanlarla ortalama düzeyde işbirliği yapabildiğini gösterir.

##### - Satış: Satış departmanı diğer departmanlarla daha güçlü bir olumlu etkileşime sahiptir (0.75). Bu, satış departmanının diğer departmanlarla daha sıkı bir işbirliği içinde olduğunu ve daha fazla sinerji yaratabileceğini gösterir.

##### - Üretim: Üretim departmanı, diğer departmanlarla olan olumlu etkileşimi düşük bir seviyede tutmaktadır (0.333). Bu, üretim departmanının diğer departmanlarla daha az işbirliği içinde olduğunu ve daha az etkileşimde bulunduğunu gösterir.

##### - Ar-Ge: Ar-Ge departmanı, diğer departmanlarla herhangi bir olumlu etkileşime sahip değildir (0.0). Bu, Ar-Ge departmanının diğer departmanlarla işbirliği yapmadığını veya etkileşimde bulunmadığını gösterir.

##### - Finans: Finans departmanı, diğer departmanlarla en güçlü olumlu etkileşime sahip olan departmandır (0.833). Bu, finans departmanının diğer departmanlarla sıkı bir işbirliği içinde olduğunu ve büyük ölçüde sinerji yarattığını gösterir.

##### ___________________________________________________________________________________________________________
##### ___________________________________________________________________________________________________________
##### ___________________________________________________________________________________________________________

In [6]:
print("\nDepartmanlar arasındaki Dematel Negatif Endeksleri:")
for i, department in enumerate(["Pazarlama", "Satış", "Üretim", "Ar-Ge", "Finans"]):
    print(f"{department}: {D_negative[i]}")


Departmanlar arasındaki Dematel Negatif Endeksleri:
Pazarlama: 0.5
Satış: 0.25
Üretim: 0.6666666666666666
Ar-Ge: 1.0
Finans: 0.16666666666666666


#### Pazarlama: 
##### - Pazarlama departmanı diğer departmanlarla orta düzeyde olumsuz etkileşime sahiptir (0.5). Bu, pazarlama departmanının diğer departmanlarla bazı olumsuz etkileşimler yaşadığını ancak genel olarak uyumlu olduğunu gösterir.

####  Satış: 
##### - Satış departmanı diğer departmanlarla daha az olumsuz etkileşime sahiptir (0.25). Bu, satış departmanının diğer departmanlarla daha az sorun yaşadığını ve uyumlu bir şekilde çalıştığını gösterir.

#### Üretim: 
##### - Üretim departmanı diğer departmanlarla daha yüksek bir olumsuz etkileşime sahiptir (0.667). Bu, üretim departmanının diğer departmanlarla daha fazla sorun yaşadığını ve uyumlu olmayabileceğini gösterir.

#### Ar-Ge: 
##### - Ar-Ge departmanı diğer departmanlarla tamamen olumsuz etkileşime sahiptir (1.0). Bu, Ar-Ge departmanının diğer departmanlarla uyumsuz olduğunu ve çatışma yaşadığını gösterir.

#### Finans:
##### -Finans departmanı diğer departmanlarla en az olumsuz etkileşime sahip olan departmandır (0.167). Bu, finans departmanının diğer departmanlarla uyumlu olduğunu ve genel olarak az sorun yaşadığını gösterir.